registration/migrations/0007_fix_invalid_domain_names.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.10.7 on 2018-01-05 17:04
+from __future__ import unicode_literals
+
+from django.db import migrations
+from django.db.models import Func, F, Value
+from django.db.models.functions import Lower
+
+
+def clean_domain_name(field_name):
+    """
+    Returns a Django query expression that replaces all underscores with a
+    hyphen, removes leading and trailing hyphens, and converts the field to
+    lower case.
+    """
+    remove_underscores = Func(
+        F(field_name),
+        Value('_'),
+        Value('-'),
+        function='replace'
+    )
+    trim_hyphens = Func(
+        remove_underscores,
+        Value('-'),
+        function='btrim',
+    )
+    remove_trailing_hyphens = Func(
```

Comment:
note: this won't strip trailing hyphens out of any `subdomain`s which created  `internal_lms_domain`, but that's probably ok.

Reply:
I think the ``remove_trailing_hyphens`` should do that.

Reply:
Yep, so it will!  My regex fu must be rusty :)